### PR TITLE
Fix and Features

### PR DESCRIPTION
**Pull request by lapolis**
Added a few minor fix and implemented 2 features. These won't be used if the json file is empty or does not exist. So no-one will be forced to use those.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,6 @@
-!template.txt
\ No newline at end of file
+!template.txt
+venv
+vonv
+vinv
+.*
+additional_config.json
```
Comment (spyr0-sec) [inline .gitignore:6]: You have this in the gitignore but also committing the file?

File: nessusToExcel.py [hunk 2]
```diff
@@ -1795,6 +1878,7 @@ def CloseExcelWriter(writer):
 parser.add_argument('--quiet', '-q', action='store_true', help='Accept defaults during execution')
 parser.add_argument('--keyword', '-k', required=False, help='Extract all information relating to this word')
 parser.add_argument('--noclean', '-n', required=False, action='store_true', help='Do not remove duplicates and merge columns in outdatedsoftware')
+# parser.add_argument('--nostyle', '-s', required=False, action='store_true', help='Do not apply any style')
```
Comment (spyr0-sec) [inline nessusToExcel.py:1881]: Do you want this commented out?

**Comment by spyr0-sec** [comment]
Thanks @lapolis, as the number of files are growing I think it would be worth moving it to a sub directory?

File: .gitignore (hunk 1 above)
Comment (lapolis) [reply .gitignore:6]: This was just so the internet can see the template but at the same time I won't be pushing my template in future commits. Does it make sense?

File: nessusToExcel.py (hunk 2 above)
Comment (lapolis) [reply nessusToExcel.py:1881]: Not yet, it's not implemented. I didn't have enough time to get to that point. Anyway if the user leaves the json file empty, most of the style is not applied anyway. I wanted to add that flag so no style at all is applied.

**Comment by lapolis** [comment]
> Thanks @lapolis, as the number of files are growing I think it would be worth moving it to a sub directory?

Yes I think it makes sense. Do you want the whole script and all config/output (hosts) in a subdir?

**Comment by spyr0-sec** [comment]
Yeah I think that makes sense, just need to make sure args.out gets updated
https://github.com/spyr0-sec/helper-scripts/blob/623b4749a1c402b9debfb2378b686986e7e646f3/nessusToExcel.py#L1915

**Comment by spyr0-sec** [comment]
I'll leave it as is for the meantime. Enjoy the CoPilot summary 😁:

This pull request introduces several changes to the `nessusToExcel.py` script and the `additional_config.json` file. The changes aim to enhance the functionality and customization of the script. The most notable changes include the addition of a new configuration file for customizing table styles, the addition of new plugin checks for Unix systems, and the modification of the `CreateExcelWriter` function to improve the appearance of the output Excel file.

Here are the most important changes grouped by theme:

### Configuration File Addition:

* [`additional_config.json`](diffhunk://#diff-b245a0152fa155373f4dd57d611396856e08ad3fdaba3d4b302927603da6de65R1-R36): The file was added to allow for more customization. It includes options for customizing the table style in the output Excel file, as well as a list of TLS issues.

### Codebase Enhancements:

* [`nessusToExcel.py`](diffhunk://#diff-31366147c269585792236bdc9e3fe4ed34b9a64adf59a87cc9a87979a606fcf6L2-R2): A JSON module was imported to allow for the reading of the new configuration file.
* [`nessusToExcel.py`](diffhunk://#diff-31366147c269585792236bdc9e3fe4ed34b9a64adf59a87cc9a87979a606fcf6R55-R63): The script now loads the additional configurations from the `additional_config.json` file.
* [`nessusToExcel.py`](diffhunk://#diff-31366147c269585792236bdc9e3fe4ed34b9a64adf59a87cc9a87979a606fcf6R1436-R1442): New plugin checks for Unix systems were added to the `extractCredPatch` function. [[1]](diffhunk://#diff-31366147c269585792236bdc9e3fe4ed34b9a64adf59a87cc9a87979a606fcf6R1436-R1442) [[2]](diffhunk://#diff-31366147c269585792236bdc9e3fe4ed34b9a64adf59a87cc9a87979a606fcf6L1440-R1458)
* [`nessusToExcel.py`](diffhunk://#diff-31366147c269585792236bdc9e3fe4ed34b9a64adf59a87cc9a87979a606fcf6L1692-R1862): The `CreateExcelWriter` function was modified to improve the appearance of the output Excel file. It now supports custom table styles and includes more detailed formatting options.
* [`nessusToExcel.py`](diffhunk://#diff-31366147c269585792236bdc9e3fe4ed34b9a64adf59a87cc9a87979a606fcf6R1881): The script now includes a new command-line argument (`--nostyle`) to control whether to apply any style.
* [`nessusToExcel.py`](diffhunk://#diff-31366147c269585792236bdc9e3fe4ed34b9a64adf59a87cc9a87979a606fcf6L1878-R1969): The script now correctly prints the path of the backup Nessus file and the output Excel file. [[1]](diffhunk://#diff-31366147c269585792236bdc9e3fe4ed34b9a64adf59a87cc9a87979a606fcf6L1878-R1969) [[2]](diffhunk://#diff-31366147c269585792236bdc9e3fe4ed34b9a64adf59a87cc9a87979a606fcf6L1968-R2052)

### Documentation Updates:

* [`nessusToExcel.py`](diffhunk://#diff-31366147c269585792236bdc9e3fe4ed34b9a64adf59a87cc9a87979a606fcf6R41-R42): The script's version was updated and a comment was added to document the addition of the Table Style customization feature and the TLS list of issues.

**Pull request merged**